### ISSUE TITLE
Enhance Apps to allow overriding properties via environment variables

### ISSUE THREAD

**Issue by john-thomas-dotcms** [dotCMS/core#25953]
We need a way to prevent dotCMS Cloud customers from changing App parameters which we, as their cloud provider, control.  On-prem customers, especially multi-tenant customers, may have similar needs to prevent accidental changes to important App parameters by dotCMS back-end users with elevated rights.

To do this, we need to enable any App parameter to be overridden by a value set by an environment variable.  The following us a suggested method to do this, but it's open to change during implementation planning.

### App Config File Changes

For each App parameter and secret, allow two additional values:
1. `envvar`: The name of an environment variable (string)
2. `envshow`: Whether to show the value of the param in the App when the param is overridden via the environment variable (true or false)

Example:
```
    param-key:
        value: "default-value"
        hidden: false
        type: "STRING"
        label: "label"
        hint: "hint"
        required: true
        envvar: "APP_ANALYTICS_CUSTOMER_KEY"
        envshow: true
```

These additional parameters should be optional, to ensure the Apps feature remains backward compatible with older App config files that don't include these values.

### App Behavior Changes

When the App starts, if the environment variable name is set for a parameter AND that environment variable is set in the environment, then the following should happen :
1. The value of the specified environment variable will be used for that App parameter.
    * Any previous value set in the App will be ignored (but won't be shown, and won't be deleted).
2. The field for that parameter will be disabled in the App.
3. If the `envshow` value is `true`, the value (from the env var) will be displayed (uneditable) in the field.
    * Otherwise, the disabled field will display the string "Set from the environment".

### Notes

* It should be possible to set any character that's valid in an App parameter via the environment variable.
    * If necessary, we should add some standard method of encoding characters in the environment variable, to ensure this.
* It should be possible to set an empty value via an environment variable.
    * In other words, there needs to be a difference between an environment variable which is not set at all, and one that's set to an empty value.
    * If the environment variable contains an empty value, the appropriate App parameter should be set to an empty value.

**Comment by john-thomas-dotcms** [comment]
We should be able to determine a standard, unique, deterministic env var name without requiring the user to specify it in the app config

**Comment by john-thomas-dotcms** [comment]
The default value for `envshow` should be true.

The exception is for app fields that are already hidden.  Currently, they're displayed with asterisks, and they should continue to be shown with asterisks whether they're set from the UX or from an env var.  But if they're set from an env var, then we should disable editing of them in the UX (so it will show the asterisks in a read-only field).

**Comment by victoralfaro-dotcms** [comment]
PR: https://github.com/dotCMS/core/pull/27705

**Comment by hmoreras** [comment]
QA video for testing: 

https://github.com/dotCMS/core/assets/31667212/e9603fce-82ea-423e-b630-99d213f46ebd

Internal QA fail: 

- The inputs are not disabled when the value is set in a environment variable. 
- Also is not respecting the case when need to show "Set from the environment"
- When the `envshow ` is false, we still show the input in the UI.

**Comment by hmoreras** [comment]
Pass intenral QA: 

<img width="1065" alt="image" src="https://github.com/dotCMS/core/assets/31667212/d5b3a2e4-f149-4d29-8185-4a4e3986efd7">


<img width="993" alt="image" src="https://github.com/dotCMS/core/assets/31667212/2d597adf-3c28-44d6-8640-53a7cecc2c79">



Note To QA: 

Variables used for testing: 

`      # Test variables
      # ANALYTICS
      DOT_APP_ANALYTICS_PARAM_CLIENT_ID: 'test-01'
      DOT_APP_ANALYTICS_PARAM_CLIENT_SECRET: 'test-02'
       #Prioroty case: 
      APP_ANALYTICS_CONFIG_URL_TEST: 'value-01'
      DOT_APP_ANALYTICS_PARAM_ANALYTICS_CONFIG_URL: 'value-01-Prioroty'

      # AMAZON TOKEN
      APP_MAX_LABELS_TEST: 'test-03'
      APP_MIN_CONFIDENCE_TEST: 'test-04'`

Note: The convention for "path varaibles" changed between the first internal QA and the sencond, from `DOT_APP_DOT_ANALYTICS_PARAM_CLIENT_ID` to `DOT_APP_ANALYTICS_PARAM_CLIENT_ID`

**Comment by josemejias11** [comment]
**Approved**: Tested on **master_c4932c7**, Docker, macOS 13.0, FF v124.0.2
